ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -737,44 +736,86 @@ impl<'a> Assemble<'a> {
 
     fn cg_loadtraceinput(&mut self, iidx: jit_ir::InstIdx, inst: &jit_ir::LoadTraceInputInst) {
         // Find the argument register containing the pointer to the live variables struct.
-        let base_reg = ARG_GP_REGS[JITFUNC_LIVEVARS_ARGIDX];
         match self.m.type_(inst.tyidx()) {
             Ty::Integer(_) | Ty::Ptr => {
+                let avoid = RegSet::from_vec(vec![
+                    Rq::RBX,
+                    Rq::R12,
+                    Rq::R13,
+                    Rq::R14,
+                    Rq::R15,
+                    Rq::RSI,
+                    Rq::RDI,
+                ]);
                 let [tgt_reg] = self.ra.get_gp_regs_avoiding(
                     &mut self.asm,
                     iidx,
                     [RegConstraint::Output],
-                    RegSet::from(base_reg),
+                    avoid,
                 );
-                let off = i32::try_from(inst.off()).unwrap();
                 let size = self.m.inst_no_proxies(iidx).def_byte_size(self.m);
                 debug_assert!(size <= REG64_SIZE);
-                match size {
-                    1 => {
-                        dynasm!(self.asm ; movzx Rq(tgt_reg.code()), BYTE [Rq(base_reg.code()) + off])
+
+                match self.m.tilocs()[usize::try_from(inst.locidx()).unwrap()] {
+                    VarLocation::Register(reg_alloc::Register::GP(reg)) => {
+                        // FIXME: Map this iidx to `reg` without this `mov`. Requires way to
+                        // initialise register allocator with existing values.
+                        dynasm!(self.asm; mov Rq(tgt_reg.code()), Rq(reg.code()));
                     }
-                    2 => {
-                        dynasm!(self.asm ; movzx Rq(tgt_reg.code()), WORD [Rq(base_reg.code()) + off])
+                    VarLocation::Register(reg_alloc::Register::FP(_)) => {
+                        panic!()
```

Comment:
I guess this is a todo rather than "something has gone wrong"?

Reply:
We know from the outermost `match` that the value is either an integer or a pointer, so I don't expect those to be stored in floating point registers. The same applies for float values which I don't expect to be stored in a normal register (see below).

Reply:
Does "expect" mean "I don't think can happen" or "I know for sure can't happen"? I think the latter. If so, is a comment worthwhile? It might be obvious enough that it's not worth doing, so please do what you see fit.

Reply:
This is the only remaining comment.

Reply:
My knowledge of x64 is still basic, so I don't know "for sure". But from what I understand the compiler shouldn't store normal integer values inside a floating point register. Added a comment in [bc99e72](https://github.com/ykjit/yk/pull/1318/commits/bc99e72e89a0b77d1d26888b4c86a8b9c266ce0f), similar to the ones I had already added for the reverse case.